glyphs-reader/src/font.rs

```diff
@@ -264,3 +294,296 @@ impl Path {
+        for name in v3_metric_names {
+            metrics.push(Plist::Dictionary(BTreeMap::from([("type".to_string(), Plist::String(name.clone()))])));
+        }        
+        self.other_stuff.insert("metrics".into(), Plist::Array(metrics));
+
+        // in each font master setup the parallel array
+        for master in self.font_master.iter_mut() {
+            let mut metric_values = Vec::new();
+            for v2_name in v2_metric_names.iter() {
+                let mut dict = BTreeMap::new();
+                if let Some(Plist::Integer(value)) = master.other_stuff.remove(v2_name) {
+                    // leave blank for 0
+                    if value != 0 {
+                        dict.insert("pos".to_string(), Plist::Integer(value));
+                    }                    
+                }
+                metric_values.push(Plist::Dictionary(dict));
+            }
+            master.other_stuff.insert("metricValues".into(), Plist::Array(metric_values));
+        }
+        Ok(())
+    }
+
+    fn v2_to_v3_shapes(&mut self) -> Result<(), Error> {
+        // shapes in v3 encompasses paths and components in v2
+        for glyph in self.glyphs.iter_mut() {
+            // v2 uses single codepoint strings, turn into int to match v3 for now
+            // In time we will likely parse unicode more carefully
+            glyph.other_stuff.entry("unicode".into()).and_modify(|v| {
+                if let Plist::String(val) = v {
+                    *v = Plist::Integer(i64::from_str_radix(val, 16).unwrap())
+                }
+            });
+
+            // Paths and components combine in shapes
+            // TODO components
+            for layer in glyph.layers.iter_mut() {
+                let paths = layer.paths.take();
+                if paths.is_some() {
+                    match &mut layer.shapes {
+                        Some(shapes) => {
+                            shapes.extend(paths.unwrap());
+                        }
+                        None => {
+                            layer.shapes = paths;
+                        }
+                    }
+                }
+            }
+        }
+        Ok(())
+    }
+
+    fn v2_to_v3_weight(&mut self, tags: &Vec<String>) -> Result<(), Error> {
+        if tags.len() != 1 || tags[0] != "wght" {
+            return Err(Error::StructuralError("Only wght supported so far".into()));
+        }
+        for master in self.font_master.iter_mut() {
+            let Some(Plist::Integer(wght)) = master.other_stuff.remove("weightValue") else {
+                continue;
+            };
+            master.other_stuff.insert("axesValues".into(), Plist::Array(vec![Plist::Integer(wght)]));
+
+            let name = match master.other_stuff.remove("weight") {
+                Some(Plist::String(name)) => name,
+                _ => weight_name(wght).to_string(),
+            };
+            master.other_stuff.insert("name".into(), Plist::String(name));
```

Comment:
so you want to derive a name for a GSFontMaster, because in Glyphs3 they have a required `name` attribute.
This would be the logic to get a master's name in a way that is compatible with both Glyphs 2 and 3:

https://github.com/googlefonts/glyphsLib/blob/f6e9c4a29ce764d34c309caef5118c48c156be36/Lib/glyphsLib/classes.py#L1637-L1644

basically in Glyphs 2, the master name was implicit in the predefined configuration of weight, width and "custom" axes (see `_joinName` method above); or it could be overridden by a user's defined custom parameter named "Master Name".